types/src/era_id.rs

```diff
@@ -0,0 +1,202 @@
+    pub fn checked_sub(&self, x: u64) -> Option<EraId> {
+        self.0.checked_sub(x).map(EraId)
+    }
+
+    /// Returns the current era minus `x`, or `0` if that would be less than `0`.
+    pub fn saturating_sub(&self, x: u64) -> EraId {
+        EraId::from(self.0.saturating_sub(x))
+    }
+
+    /// Returns whether this is era 0.
+    pub fn is_genesis(&self) -> bool {
+        self.0 == 0
+    }
+
+    /// Returns little endian bytes.
+    pub fn to_le_bytes(self) -> [u8; 8] {
+        self.0.to_le_bytes()
+    }
+
+    /// Creates new [`EraId`] instance from little endian bytes.
+    pub(crate) fn from_le_bytes(bytes: [u8; 8]) -> EraId {
+        EraId::from(u64::from_le_bytes(bytes))
+    }
+
+    /// Returns a raw value held by this [`EraId`] instance.
+    ///
+    /// You should prefer [`From`] trait implementations over this method where possible.
+    pub fn value(self) -> u64 {
+        self.0
+    }
+}
+
+impl FromStr for EraId {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        u64::from_str(s).map(EraId)
+    }
+}
+
+impl Add<u64> for EraId {
+    type Output = EraId;
+
+    fn add(self, x: u64) -> EraId {
+        EraId::from(self.0 + x)
+    }
+}
+
+impl AddAssign<u64> for EraId {
+    fn add_assign(&mut self, x: u64) {
+        self.0 += x;
+    }
+}
+
+impl Sub<u64> for EraId {
+    type Output = EraId;
+
+    fn sub(self, x: u64) -> EraId {
+        EraId::from(self.0 - x)
+    }
+}
+
+impl Display for EraId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "era {}", self.0)
+    }
+}
+
+impl From<EraId> for u64 {
+    fn from(era_id: EraId) -> Self {
+        era_id.value()
+    }
+}
+
+impl From<u64> for EraId {
+    fn from(era_id: u64) -> Self {
+        EraId(era_id)
+    }
+}
+
+impl ToBytes for EraId {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        self.0.to_bytes()
+    }
+
+    fn serialized_length(&self) -> usize {
+        self.0.serialized_length()
+    }
+}
+
+impl FromBytes for EraId {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let (id_value, remainder) = u64::from_bytes(bytes)?;
+        let era_id = EraId::from(id_value);
+        Ok((era_id, remainder))
+    }
+}
+
+impl CLTyped for EraId {
+    fn cl_type() -> CLType {
+        CLType::U64
+    }
+}
+
+impl Distribution<EraId> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> EraId {
+        EraId(rng.gen())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use alloc::collections::BTreeSet;
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::gens::era_id_arb;
+
+    #[test]
+    fn should_calculate_correct_inclusive_future_eras() {
+        let auction_delay = 3;
+
+        let current_era = EraId::from(42);
+
+        let window: BTreeSet<EraId> = current_era.iter_inclusive(auction_delay).collect();
```

Comment:
The iterator's values should already be ordered:
```suggestion
        let window: Vec<EraId> = current_era.iter_inclusive(auction_delay).collect();
```